### DISCUSSION TITLE
Open livechat programmatically

### DISCUSSION THREAD

**Discussion by softmarshmallow** [chatwoot/chatwoot#4760] [Q&A] [answered]
We can add chat button via below script, but what I want is to have my own support button that has "Chat with us" menu in it.
-> How would I be able to control the "chat-open" action via script? (and hide the built-in ui)

```html
    <script>
      (function(d,t) {
        var BASE_URL="https://app.chatwoot.com";
        var g=d.createElement(t),s=d.getElementsByTagName(t)[0];
        g.src=BASE_URL+"/packs/js/sdk.js";
        g.defer = true;
        g.async = true;
        s.parentNode.insertBefore(g,s);
        g.onload=function(){
          window.chatwootSDK.run({
            websiteToken: 'xxxxxxxx',
            baseUrl: BASE_URL
          })
        }
      })(document,"script");
    </script>
```

**Comment by pranavrajs** [comment, marked as answer]
Please see this guide. https://www.chatwoot.com/docs/product/channels/live-chat/sdk/setup#trigger-widget-without-displaying-bubble

**Reply by CelestineAkpanoko** [reply]
Hi @pranavrajs, I hope you are well. Using the chatwootSDK, how can I get the chatbot to send a message first after the user clicks on the widget, and also how I can get make the `Email Collect Prompt` compulsory for the user.

**Reply by pranavrajs** [reply]
@CelestineAkpanoko Please use [Campaigns](https://www.chatwoot.com/docs/product/features/campaigns/ongoing) for outbound messages and [Pre-chat Form](https://www.chatwoot.com/docs/user-guide/features/pre-chat-form) to make it mandatory for a user to share the email.

**Reply by CelestineAkpanoko** [reply]
@pranavrajs, thanks for your reply. But I am avoiding the pre-chat form because it looks static. Is it possible to make it mandatory for the user to input their email by the `Email Collect Prompt` or by a setting in within the ChatwootSDK?

**Comment by softmarshmallow** [comment]
For future reference. Implemented on react / nextjs
> https://github.com/gridaco/grida/pull/42
